source/CPluginSystem.cpp

```diff
@@ -68,14 +68,23 @@ void CPluginSystem::LoadPlugins()
     {
         for (auto it = paths.crbegin(); it != paths.crend(); it++)
         {
-            const auto filename = it->c_str();
+            std::string filename = *it;
+
+            // ModLoader support: keep game dir relative paths relative
+            if (filename.length() > Filepath_Game.length() && // and separator
+                filename[Filepath_Game.length()] == '\\' && // path separator after game path
+                StringStartsWith(filename, Filepath_Game, false)) // path is within game root
+            {
+                filename.replace(0, Filepath_Game.length() + 1, ""); // remove game root path prefix
+            }
```

Comment:
```
 if  StringStartsWith(filename, Filepath_Game + "\\", false)) // path is within game root
            {
                filename.replace(0, Filepath_Game.length() + 1, ""); // remove game root path prefix
            }
```

can be done like this?

Reply:
It will be fooled with **C:\Game_Dir_plus_Some_Other** which will be shortened to **_plus_Some_Other**

Reply:
> Filepath_Game + "\\"

Reply:
Right. One downside it is now copying whole string.

Reply:
Big deal?

Reply:
If you worry about performance, create a separate static variable with trailing slash

Reply:
Ok I added generic purpose util function for that.

Reply:
still think that this is much more readable

```c++
    // strip parent prefix from filepath if present
    static void FilepathRemoveParent(std::string& path, const std::string_view base)
    {
        if (StringStartsWith(path, base + '\\', false)) {
            path.replace(0, base.length() + 1, "");
        }
    }
```

Reply:
Will 

> still think that this is much more readable
> 
> ```c++
>     // strip parent prefix from filepath if present
>     static void FilepathRemoveParent(std::string& path, const std::string_view base)
>     {
>         if (StringStartsWith(path, base + '\\', false)) {
>             path.replace(0, base.length() + 1, "");
>         }
>     }
> ```

Will not work in case where path is just base path.

Reply:
Makes sense, ok.